lxd/device/device.go

```diff
@@ -0,0 +1,128 @@
+package device
+
+import (
+	"fmt"
+
+	"github.com/lxc/lxd/lxd/device/config"
+	"github.com/lxc/lxd/lxd/state"
+)
+
+// devTypes defines supported top-level device type creation functions.
+var devTypes = map[string]func(config.Device) device{
+	"nic": loadNICByType,
```

Comment:
Should we put in an exported function that lets you list all the valid devTypes?
Might be useful when doing some config validation or at other spots in the code.

Reply:
@stgraber sounds good, will add and hook up to type validation.

Reply:
@stgraber actually looking at this now, I can't find a use for this at the current time (nictype is validated just like any other config key by calling `d.validate()` as part of `New()`. Can we leave this out until we need it (perhaps for an API that lists valid types?), unless you can see a use case I've missed?